va/va.go

```diff
@@ -374,55 +375,37 @@ func (va *ValidationAuthorityImpl) validateChallenge(ctx context.Context, identi
 	ctx context.Context,
-	domain string,
-	challenge core.Challenge,
-	authz core.Authorization,
+	req *vapb.PerformValidationRequest,
 	results chan *remoteValidationResult) {
 	for _, i := range rand.Perm(len(va.remoteVAs)) {
 		remoteVA := va.remoteVAs[i]
 		go func(rva RemoteVA, index int) {
-			_, err := rva.PerformValidation(ctx, domain, challenge, authz)
-			if err != nil {
-				// returned error can be a nil *probs.ProblemDetails which breaks the
-				// err != nil check so do a slightly more complicated unwrap check to
-				// make sure we don't choke on that.
-				// TODO(@cpu): Clean this up once boulder issue 2254[0] is resolved
-				// [0] https://github.com/letsencrypt/boulder/issues/2254
-				if p, ok := err.(*probs.ProblemDetails); ok && p != (*probs.ProblemDetails)(nil) {
-					// If the non-nil err was a non-nil *probs.ProblemDetails then we can
-					// log it at an info level. It's a normal non-success validation
-					// result and the remote VA will have logged more detail.
-					va.log.Infof("Remote VA %q.PerformValidation returned problem: %s", rva.Address, err)
-				} else if ok && p == (*probs.ProblemDetails)(nil) {
-					// If the non-nil err was a nil *probs.ProblemDetails then we don't need to do
-					// anything. There isn't really an error here.
-					err = nil
-				} else if canceled.Is(err) {
-					// If the non-nil err was a canceled error, ignore it. That's fine it
-					// just means we cancelled the remote VA request before it was
-					// finished because we didn't care about its result.
-					err = nil
-				} else if !ok {
-					// Otherwise, the non-nil err was *not* a *probs.ProblemDetails and
-					// was *not* a context cancelleded error and represents something that
-					// will later be returned as a server internal error
-					// without detail if the number of errors is >= va.maxRemoteFailures.
-					// Log it at the error level so we can debug from logs.
-					va.log.Errf("Remote VA %q.PerformValidation failed: %s", rva.Address, err)
-				}
-			}
 			result := &remoteValidationResult{
 				VAHostname: rva.Address,
 			}
-			if err == nil {
-				results <- result
-			} else if prob, ok := err.(*probs.ProblemDetails); ok {
-				result.Problem = prob
-				results <- result
-			} else {
-				result.Problem = probs.ServerInternal("Remote PerformValidation RPC failed")
-				results <- result
+			res, err := rva.PerformValidation(ctx, req)
+			if err != nil && canceled.Is(err) {
+				// If the non-nil err was a canceled error, ignore it. That's fine: it
+				// just means we cancelled the remote VA request before it was
+				// finished because we didn't care about its result.
+				err = nil
+			} else if err != nil {
+				// This is a real error, not just a problem with the validation.
+				va.log.Errf("Remote VA %q.PerformValidation failed: %s", rva.Address, err)
+				result.Problem = probs.ServerInternal(
+					fmt.Sprintf("Remote PerformValidation RPC failed: %s", err))
```

Comment:
Since `Problem` is exposed to the user, we tend to be conservative about what we include: We try to only interpolate errors if we are confident that they contain things that are useful to the user / don't expose details of our infrastructure unintentionally. The previous code here just used `probs.ServerInternal("Remote PerformValidation RPC failed")`; I think we should keep that.

Reply:
Makes sense! Reverted this message and the test which checks for it.